src/taipy/core/data/mongo.py

```diff
@@ -0,0 +1,207 @@
+        parent_id: Optional[str] = None,
+        last_edit_date: Optional[datetime] = None,
+        job_ids: List[JobId] = None,
+        cacheable: bool = False,
+        validity_period: Optional[timedelta] = None,
+        edit_in_progress: bool = False,
+        properties: Dict = None,
+    ):
+        if properties is None:
+            properties = {}
+        required = self._REQUIRED_PROPERTIES
+        if missing := set(required) - set(properties.keys()):
+            raise MissingRequiredProperty(
+                f"The following properties " f"{', '.join(x for x in missing)} were not informed and are required"
+            )
+
+        self._check_exposed_type(properties[self.__CUSTOM_DOCUMENT_PROPERTY])
+
+        super().__init__(
+            config_id,
+            scope,
+            id,
+            name,
+            parent_id,
+            last_edit_date,
+            job_ids,
+            cacheable,
+            validity_period,
+            edit_in_progress,
+            **properties,
+        )
+
+        mongo_client = _connect_mongodb(
+            db_host=properties.get("db_host", "localhost"),
+            db_port=properties.get("db_port", 27017),
+            db_username=properties.get("db_username"),
+            db_password=properties.get("db_password"),
+        )
+        self.collection = mongo_client[properties.get("db_name")][properties.get("collection_name")]
+
+        self.custom_document = properties[self.__CUSTOM_DOCUMENT_PROPERTY]
+
+        self._decoder = self._default_decoder
+        custom_decoder = getattr(self.custom_document, "decode", None)
+        if callable(custom_decoder):
+            self._decoder = custom_decoder
+        else:
+            self._decoder = self._default_decoder
+
+        self._encoder = self._default_encoder
+        custom_encoder = getattr(self.custom_document, "encode", None)
+        if callable(custom_encoder):
+            self._encoder = custom_encoder
+        else:
+            self._encoder = self._default_encoder
+
+        if not self._last_edit_date:
+            self.unlock_edit()
+
+    def _check_exposed_type(self, exposed_type):
+        if not isclass(exposed_type):
+            raise InvalidExposedType(f"Invalid exposed type of {exposed_type}. Only custom object class are supported.")
+
+    @classmethod
+    def storage_type(cls) -> str:
+        return cls.__STORAGE_TYPE
+
+    def _read(self):
+        cursor = self._read_by_query()
+
+        return list(map(lambda row: self._decoder(row), cursor))
+
+    def _read_by_query(self):
+        """Query from a Mongo collection, exclude the _id field"""
+
+        return self.collection.find(self.read_query)
+
+    def _write(self, data) -> None:
+        """Check data against a collection of types to handle insertion on the database."""
+
+        if not isinstance(data, list):
+            data = [data]
+
+        if len(data) == 0:
+            self.collection.delete_many({})
```

Comment:
I guess using the drop command is better in terms of performance. 
```suggestion
            self.collection.drop()
```
However, I wonder if we need to recreate the collection after dropping it.
`self.collection = mongo_client[properties.get("db_name")][properties.get("collection_name")]`

Reply:
You are right on `.drop()` method.

The recreation of a collection is actually not necessary, since if a collection is not exist, the `find()` method will return a empty cursor, which can be converted to an empty list.